File: Dreya/torrent/Series.java
package dreya.torrent;

import dreya.Config;
import dreya.Database;
import dreya.messenger.Sender;
import java.io.FileOutputStream;
import java.io.IOException;
import java.net.URL;
import java.nio.channels.Channels;
import java.nio.channels.ReadableByteChannel;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

/**
 *
 * @author smv
 */
public class Series implements java.io.Serializable  {
    
    private String name = "";
    private String torrentTitle = "";
    private String subtitleText = ""; 
    private String notIn = "";
    
    public int episode = 0;
    public int season = 0;
    
    public Series(String name, String torrentTitle, String subtitleText, String notIn)
    {
        this.name = name;
        this.torrentTitle = torrentTitle;
        this.subtitleText = subtitleText;
        this.notIn = notIn;  
        
        this.episode = Integer.parseInt(Database.get(this.name+".episode", "0"));
        this.season = Integer.parseInt(Database.get(this.name+".season", "0"));
    }
    
    public String getName()
    {
        return this.name;
    }
    
    public String getSubtitleText()
    {
        return this.subtitleText;
    }
    
    
    public String getTorrentTitle()
    {
        return this.torrentTitle;
    }
    
    public void check(String title, String url)
    {
        title = title.toLowerCase();
        
        
        if ((title.contains(this.torrentTitle)) & (title.contains("720p")))
        {
            if (("".equals(this.notIn)) | (!title.contains(this.notIn)))
            {
                this.download(url);
            }
        }
    }
    
    
    public void download(String url)
    {
        
        String[] urlParts = url.split("/");
        String fileName = urlParts[urlParts.length-1];
        
        System.setProperty("http.agent", "Mozilla/5.0 (Android 4.4; Mobile; rv:41.0) Gecko/41.0 Firefox/41.0"); 
        
        Pattern MY_PATTERN = Pattern.compile("S[0-9].E[0-9].");
        Matcher m = MY_PATTERN.matcher(fileName);
        
        if (m.find())
        {
            String seepcode = m.group();
            seepcode = seepcode.toLowerCase();
            seepcode = seepcode.replace("s","");
            
            String[] parts = seepcode.split("e");
            
            int testSeason = Integer.parseInt(parts[0]);
            int testEpisode = Integer.parseInt(parts[1]);
            
            if ((testSeason>this.season) | ((testSeason==this.season) & (testEpisode>this.episode))) { 
                try
                {
                    
                    URL website = new URL(url);
                    ReadableByteChannel rbc = Channels.newChannel(website.openStream());
                    FileOutputStream fos = new FileOutputStream(Config.showsTorrentFolder+fileName);
                    fos.getChannel().transferFrom(rbc, 0, Long.MAX_VALUE);        
                    
                    this.episode = testEpisode;
                    this.season = testSeason;
                    
                    
                    Database.save(this.name+".episode", this.episode+"");
                    Database.save(this.name+".season", this.season+"");
                    Database.store();
                          
                    String d_ep = this.episode+"";
                    if (this.episode<10)
                    {
                        d_ep = "0"+this.episode;
                    }
                    String d_se = this.season+"";
                    if (this.season<10)
                    {
                        d_se = "0"+this.season;
                    }
                    
                    
                    Sender.send("", this.getName()+" "+d_se+"x"+d_ep+" rész letöltve!");
                    
                    System.out.println("[info] Letöltve: "+url);
                    
                }
                catch (IOException ex) {
                    System.out.println("[error] "+ex.getMessage());
                }            
            }
            
        }
    }
    
}

File: Dreya/Series/Series.py
# -*- coding: utf-8 -*-

from Messenger.Messenger import Messenger
from Http import Http
from Terminal import Terminal
import re

class Series():

    season = 0
    episode = 0

    name = ""
    torrent = ""
    subtitle = ""
    notIn = ""

    cacheKey = ""

    def __init__(self, name, torrent, subtitle, notIn):
        self.name = name
        self.cacheKey = "series."+name

        if name:
            self.torrent = torrent.lower()
        if subtitle:
            self.subtitle = subtitle.lower()
        if notIn:
            self.notIn = notIn.lower()

        from Database import Database

        self.episode = int(Database.get(self.cacheKey+".episode", "0"))
        self.season = int(Database.get(self.cacheKey + ".season", "0"))


    def checkSubtitle(self, text):
        text = text.lower()
        if self.subtitle in text:
            return True

        return False


    def checkDownload(self, text, url):
        text = text.lower()

        from Config import Config
        from Database import Database


        if self.torrent in text and "720p" in text and ((self.notIn and self.notIn not in text) or self.notIn==""):

            output = re.search('s[0-9].e[0-9].', text, flags=re.IGNORECASE).group(0)
            ep = re.findall('[0-9].', output, flags=re.IGNORECASE)
            season = int(ep[0])
            episode = int(ep[1])


            if (season>self.season | (season==self.season & episode>self.episode)):

                filename = url.split("/")
                filename = filename[-1]

                try:
                    Http.download(url, Config.showsTorrentFolder+filename)

                    self.season = season
                    self.episode = episode


                    Database.set(self.cacheKey + ".episode", episode)
                    Database.set(self.cacheKey + ".season", season)

                    Messenger.sendMessage("", self.name + " S" + ep[0] + "E" + ep[1] + " torrent letöltve!")
                    Terminal.info(self.name + " S" + ep[0] + "E" + ep[1] + " torrent letöltve!", "info")

                except IOError as e:
                    Terminal.error("", "series.error", e)